openwisp_notifications/api/views.py

```diff
@@ -0,0 +1,67 @@
+import uuid
+
+from django.utils.translation import gettext_lazy as _
+from django_filters.rest_framework import DjangoFilterBackend
+from openwisp_notifications.api.serializers import NotificationSerializer
+from openwisp_notifications.swapper import load_model
+from rest_framework import status
+from rest_framework.authentication import BasicAuthentication, SessionAuthentication
+from rest_framework.generics import GenericAPIView
+from rest_framework.mixins import ListModelMixin
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import DjangoModelPermissions
+from rest_framework.response import Response
+
+Notification = load_model('Notification')
+
+
+class NotificationPaginator(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = 'page_size'
+    max_page_size = 100
+
+
+class BaseNotificationView(GenericAPIView):
+    model = Notification
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    permission_classes = [DjangoModelPermissions]
+    queryset = Notification.objects.all()
+    serializer_class = NotificationSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(recipient=self.request.user)
+
+
+class NotificationListView(BaseNotificationView, ListModelMixin):
+    pagination_class = NotificationPaginator
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = ['unread']
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+
+class NotificationReadView(BaseNotificationView):
```

Comment:
I would call this `NotificationDetailView`

Reply:
This view does not provide details about a particular notification. It just mark them as read.

Reply:
I understand, I misread this.

Marking notifications as read does not need to return anything, because the consumer of the API is supposed to already have the list of notifications and we don't need to send back anything.

I would do the following:

```
`GET /api/v1/notifications/` to get the list of notifications
`GET /api/v1/notifications/<pk>/` to get the details of a specific notification, we add this just to make the API more complete
`POST /api/v1/notifications/read/` to flag all notifications as read, make it return just 200 OK and empty response body.
```